### PR TITLE
fix arg parsing testing

### PR DESCRIPTION
**Pull request by kbknapp**
This PR extends the UI testing.

Prior to this PR the UI testing stopped at clap building the argument definitions, but that's only half the use. We were seeing errors in the manner that we _used_ these values. This PR extends the testing into our use of the CLI values such that it now stops _after_ our normalization step where all the errors were found.

This works by doing the same `CliCommand::traverse_exec` traversal, except that it skips the `CliCommand::run` step so that nothing is actually run.

There are a few areas where the `CliCommand::update_ctx` (which does our CLI value normalization) _did_ do some I/O. Those expressions have been `cfg`'ed out during tests.

### CODE REVIEW

**Comment by kbknapp** [comment]
Current dependencies on/for this PR:
* main
  * **PR #282** <a href="https://app.graphite.dev/github/pr/seaplane-io/seaplane/282" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/seaplane-io/seaplane/282?utm_source=stack-comment).

**Review by ericsampson — APPROVED** [review]
What about doing one of these two approaches, to avoid having the same cfg predicates pasted in a ton of places?
https://stackoverflow.com/a/62984789

File: seaplane-cli/src/cli/cmds/metadata/set.rs [hunk 1]
```diff
@@ -52,6 +52,8 @@ impl CliCommand for SeaplaneMetadataSet {
     }
 
     fn update_ctx(&self, matches: &ArgMatches, ctx: &mut Ctx) -> Result<()> {
+        // NOTE: MetadataCtx::from_md_set is inpure and tries to read from STDIN if the value is
```
Comment (chrisdickinson) [inline seaplane-cli/src/cli/cmds/metadata/set.rs:55]:
```suggestion
        // NOTE: MetadataCtx::from_md_set is impure and tries to read from STDIN if the value is
```

(sorry, sorry, I can't help myself! 😓 )

File: seaplane-cli/src/lib.rs [hunk 2]
```diff
@@ -35,16 +35,41 @@ pub use crate::{
...
+    pub fn test_main_update_ctx(matches: &ArgMatches) -> Result<()> {
+        let mut ctx = Ctx::default();
+        let s: Box<dyn CliCommand> = Box::new(Seaplane);
+        s.traverse_update_ctx(&matches, &mut ctx)?;
+        Ok(())
+    }
+
+    pub fn test_main_exec_with_ctx(matches: &ArgMatches, mut ctx: Ctx) -> Result<()> {
+        let s: Box<dyn CliCommand> = Box::new(Seaplane);
+        s.traverse_exec(&matches, &mut ctx)?;
```
Comment (chrisdickinson) [inline seaplane-cli/src/lib.rs:69]: (I peeked into `traverse_exec` and found a much more advanced use of Clap than I'm used to! TIL!)

**Review by chrisdickinson — APPROVED** [review]
LGTM!

File: seaplane-cli/src/cli/cmds/metadata/set.rs (hunk 1 above)
Comment (kbknapp) [reply seaplane-cli/src/cli/cmds/metadata/set.rs]: 😅

**Pull request merged**